### PR TITLE
Add a statistics box

### LINKED ISSUE #104: View statistics of appointment book
Check the number of upcoming/done appointments on today / in this week.

### PR DESCRIPTION
**Pull request by YangYue128-helen**
Now you can view the number of done/ upcoming appointments for today/this week in the UI.

### CODE REVIEW

File: src/main/java/seedu/address/model/AppointmentBook.java [hunk 1]
```diff
@@ -142,6 +145,24 @@ public String toString() {
         return appointments.asUnmodifiableObservableList();
     }
 
+    @Override
+    public AppointmentStatistics getAppointmentBookStatistics() {
+        Date td = new Date(LocalDate.now());
```
Comment (xz0127) [inline src/main/java/seedu/address/model/AppointmentBook.java]: Would it be clearer to change the variable name `td` to something more concrete, maybe `todayDate`?

File: src/main/java/seedu/address/model/AppointmentStatistics.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Returns the number of upcoming appointments in today.
+     */
+    public int getUpcomingToday() {
+        return this.upcomingToday;
+    }
+
+    /**
+     * Returns the number of done appointments in today.
```
Comment (xz0127) [inline src/main/java/seedu/address/model/AppointmentStatistics.java]: Would it be better to change the `done appointments` to `appointments done`? `appointment done` might sound smoother in terms of grammar? Let me know what do you think. 👍

File: src/main/java/seedu/address/model/AppointmentStatistics.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * Collects related statistics of appointment book.
+ */
+public class AppointmentStatistics {
+    private int doneToday;
+    private int upcomingToday;
+    private int doneThisWeek;
+    private int upcomingThisWeek;
+
+    /**
+     * Creates a AppointmentStatistics with the {@code totalToday}, {@code doneToday}
```
Comment (xz0127) [inline src/main/java/seedu/address/model/AppointmentStatistics.java]: Might consider changing `a` into `an`.

File: src/main/java/seedu/address/model/AppointmentStatistics.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Returns the number of upcoming appointments in this week.
+     */
+    public int getUpcomingThisWeek() {
+        return this.upcomingThisWeek;
+    }
+
+    /**
+     * Returns the number of done appointments in this week.
```
Comment (xz0127) [inline src/main/java/seedu/address/model/AppointmentStatistics.java]: Same as the comment mentioned above.

File: src/main/java/seedu/address/model/AppointmentStatistics.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package seedu.address.model;
+
+/**
+ * Collects related statistics of appointment book.
+ */
+public class AppointmentStatistics {
+    private int doneToday;
```
Comment (xz0127) [inline src/main/java/seedu/address/model/AppointmentStatistics.java]: It is not a big issue but personally I feel the naming of the attributes can be clearer? Let me know what do you think. 👍

File: src/main/java/seedu/address/model/ReadOnlyAppointmentBook.java [hunk 6]
```diff
@@ -14,4 +14,8 @@
      */
     ObservableList<Appointment> getAppointmentList();
 
+    /**
+     * Returns the statistics of the appointment books.
```
Comment (xz0127) [inline src/main/java/seedu/address/model/ReadOnlyAppointmentBook.java]: Would singular form `book` be better?

File: src/main/java/seedu/address/model/appointment/Date.java [hunk 7]
```diff
@@ -62,6 +65,20 @@ public boolean isAfter(Date inputDate) {
         return value.isAfter(inputDate.value);
     }
 
+    /**
+     * Checks if the appointment date is in the same week with the input {@code Date}.
+     *
+     * @return true if the appointment date is in same week with the input date, false otherwise.
+     */
+    public boolean isInSameWeek(Date inputDate) {
+        TemporalAdjuster adj = TemporalAdjusters.next(DayOfWeek.MONDAY);
```
Comment (xz0127) [inline src/main/java/seedu/address/model/appointment/Date.java]: Would it be better if the variable name `adj` is changed into something clearer?

File: src/test/java/seedu/address/model/AppointmentStatisticsTest.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class AppointmentStatisticsTest {
+    @Test
+    public void toStringBasedOn() {
+        AppointmentBook b = new AppointmentBook();
```
Comment (xz0127) [inline src/test/java/seedu/address/model/AppointmentStatisticsTest.java]: Would it be better to rename the variable `b` to make it clearer?

**Review by xz0127 — COMMENTED** [review]
Other than the comments mentioned, the rest LGTM!! 👍

**Review by JinHao-L — APPROVED** [review]
LGTM! 👍

**Pull request merged**